install/0000_00_cluster-version-operator_02_usc_rbac-DevPreviewNoUpgrade.yaml

```diff
@@ -0,0 +1,165 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: update-status-controller-library
+  namespace: openshift-cluster-version
+  annotations:
+    kubernetes.io/description: Role that allows the USC binary to operate local resources needed by the o/library-go controller code
+    include.release.openshift.io/self-managed-high-availability: "true"
+    release.openshift.io/feature-set: DevPreviewNoUpgrade
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - list
+- apiGroups:
+  - ""
+  resources:
+  - events
+  verbs:
+  - create
+- apiGroups:
+  - apps
+  resources:
+  - replicasets
+  verbs:
+  - get
+- apiGroups:
+  - coordination.k8s.io
+  resources:
+  - leases
+  verbs:
+  - get
+  - create
+  - update
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: update-status-controller-library
+  namespace: openshift-cluster-version
+  annotations:
+    kubernetes.io/description: Role that allows the USC binary to operate cluster resources needed by the o/library-go controller code
+    include.release.openshift.io/self-managed-high-availability: "true"
+    release.openshift.io/feature-set: DevPreviewNoUpgrade
+rules:
+- apiGroups:
+  - config.openshift.io
+  resources:
+  - infrastructures
```

Comment:
This and `replicasets` in the Role are weird permissions.  Worth pushing back on the library-go controller maintainers about making it possible to reduce this access in controllers who don't (as far as I can tell) actually need to depend on whatever functionality _does_ need access to these resources?  Or just leave it alone?  Either way, not a blocker for merging this scaffolding; we can always adjust later if library-go does make changes.

Reply:
I have not investigated this, just added anything that the controller actually failed on startup. The library code contains more controllers that handle some stuff behind the scenes, but I have not looked at the details (I also have not found any docs besides godocs for that library). I will make a note to investigate this - it would be good to have a better understanding what we are pulling in with the library-go code.

But I would not block on it.